bentoml/_internal/utils/circus.py

```diff
@@ -21,10 +33,10 @@ def create_circus_socket_from_uri(
     *args: t.Any,
     name: str = "",
     **kwargs: t.Any,
-) -> CircusSocket:
-    from circus.sockets import CircusSocket  # type: ignore[reportMissingTypeStubs]
+) -> CircusSocket | t.NoReturn:  # pragma: no cover (test coverred by circus)
```

Comment:
Isn't `NoReturn` for when a function will *never* return?

Reply:
`NoReturn` is for function when it raises an exception

Reply:
@aarnphm actually not. Every function may raise exceptions. `NoRetuen` means a function never returns. Constantly raising an exception is just one of the cases. The case hers should not use `NoReturn`

Reply:
Got it let me update accordingly